classes/Member.php

```diff
@@ -142,6 +142,48 @@ public function image() {
 
     }
 
+    // TODO: some of this probably wants to go elsewhere
+    public function getMostRecentMembership() {
+        $last_cons = '';
+        $last_house = null;
+        $last_party = null;
+        if ( $this->current_member_anywhere() ) {
```

Comment:
I don't follow the logic here, I'm afraid. `search_member_db_lookup` returns you the min/max entered/left house which could be used directly (or if not, that should probably change also). If you just have a Member, then `$this->house_disp` and the direct `$this->party` and `->constituency` should give you the "current" (display) details.